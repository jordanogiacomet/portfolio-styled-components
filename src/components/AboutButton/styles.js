import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
