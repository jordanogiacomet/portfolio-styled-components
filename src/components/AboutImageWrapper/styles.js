import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
