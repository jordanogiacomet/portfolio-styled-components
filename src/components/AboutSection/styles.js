import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
