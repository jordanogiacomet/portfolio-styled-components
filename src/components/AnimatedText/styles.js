import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;