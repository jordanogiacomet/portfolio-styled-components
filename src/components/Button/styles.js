import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 