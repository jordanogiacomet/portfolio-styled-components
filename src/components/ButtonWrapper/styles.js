import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;