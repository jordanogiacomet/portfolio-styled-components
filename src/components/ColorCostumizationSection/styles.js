// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
