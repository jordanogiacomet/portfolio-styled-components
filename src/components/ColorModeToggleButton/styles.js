import styled from 'styled-components';

export const Container = styled.button`
  padding: 1rem;
  background-color: transparent;
  box-shadow: none;
  border: 2px solid #ADB7BE;
  border-radius: 9999px;
  color: #ADB7BE;
  font-size: 1.2rem; 
  right: 10rem;
  cursor: pointer;
  z-index: 199999;
  transition: 0.2s;
  &:hover{
    color: #4169E1;
    border-color: #4169E1;
  }
  @media(max-width: 768px) {
    bottom: 20px;
  }
  &.light{
    color: white;
    border-color: white;
    &:hover{
      color: #CCCCCC;
      border-color: #CCCCCC;
    }
  }

`;

