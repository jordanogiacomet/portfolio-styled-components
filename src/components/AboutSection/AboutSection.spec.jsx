import { renderTheme } from '../../styles/render-theme';
import { AboutSection } from '.';
import { screen, fireEvent } from '@testing-library/react';
import { Footer } from '../Footer';

describe('<AboutSection />', () => {
  it('should render AboutSection with English language', () => {
    renderTheme(
      <>
        <AboutSection />
        <Footer />
      </>
    );

    const button = screen.getByText('PT-BR');
    fireEvent.click(button);
    expect(screen.getByText('About me')).toBeInTheDocument();
  });

  it('should render AboutSection with default language', () => {
    renderTheme(<AboutSection />);
    expect(screen.getByText('Sobre mim')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<AboutSection />);
    expect(container).toMatchSnapshot();
  });
});
