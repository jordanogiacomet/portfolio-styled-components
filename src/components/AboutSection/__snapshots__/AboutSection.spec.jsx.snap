// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutSection /> should match snapshot 1`] = `
.c5 {
  color: #ADB7BE;
  font-size: 3.2rem;
  text-transform: none;
  z-index: 2;
}

.c6 {
  font-size: 1.5rem;
  color: #FFFFFF;
  margin-bottom: 30px;
  margin-top: 30px;
}

.c8 {
  background: transparent;
  position: relative;
  border: none;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  color: #FFFFFF;
  color: #1E90FF;
}

.c8.light {
  color: black;
  color: #1E90FF;
}

.c8.light::after {
  width: 100%;
}

.c8:hover {
  color: #1E90FF;
}

.c8::after {
  content: '';
  display: block;
  height: 2px;
  width: 0;
  position: absolute;
  left: 0;
  bottom: 0;
  transition: width 0.3s ease-in-out;
  background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
}

.c8:hover::after {
  width: 100%;
}

.c8::after {
  width: 100%;
}

.c9 {
  background: transparent;
  position: relative;
  border: none;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  color: #FFFFFF;
}

.c9.light {
  color: black;
}

.c9:hover {
  color: #1E90FF;
}

.c9::after {
  content: '';
  display: block;
  height: 2px;
  width: 0;
  position: absolute;
  left: 0;
  bottom: 0;
  transition: width 0.3s ease-in-out;
  background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
}

.c9:hover::after {
  width: 100%;
}

.c10 {
  margin-top: 8px;
  color: #FFFFFF;
}

.c11 {
  list-style-type: disc;
  padding-left: 0.5rem;
}

.c11 >li {
  margin-top: 8px;
}

.c11.light >li {
  color: black;
}

.c2 {
  width: 100%;
  height: auto;
}

.c4 {
  background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
  -webkit-background-clip: text;
  -webkit-text-fill-color: transparent;
}

.c0 {
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
}

.c1 {
  margin-top: 4px;
  order: 2;
}

.c1 >img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px;
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1),0 6px 20px rgba(0, 0, 0, 0.1);
}

.c3 {
  margin-top: 30px;
  margin-bottom: 30px;
  text-align: left;
  display: flex;
  flex-direction: column;
  order: 1;
  height: auto;
}

.c7 {
  display: flex;
  flex-direction: row;
  margin-top: 8px;
}

@media (min-width: 640px) {
  .c6 {
    font-size: 1.8rem;
  }
}

@media (min-width: 1024px) {
  .c6 {
    font-size: 2rem;
  }
}

@media (min-width: 768px) {
  .c8 {
    font-size: 2rem;
  }
}

@media (max-width: 640px) {
  .c8 {
    font-size: 1.6rem;
  }
}

@media (min-width: 768px) {
  .c9 {
    font-size: 2rem;
  }
}

@media (max-width: 640px) {
  .c9 {
    font-size: 1.6rem;
  }
}

@media (max-width: 640px) {
  .c11 {
    font-size: 1.5rem;
  }
}

@media (min-width: 768px) {
  .c0 {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
}

@media (min-width: 1280px) {
  .c0 {
    gap: 50px;
  }
}

@media (min-width: 768px) {
  .c1 {
    margin-top: 0;
  }
}

@media (min-width: 768px) {
  .c3 {
    margin-top: 0;
  }
}

<div>
  <div
    id="about"
  >
    <section
      class="c0"
      type="about-section"
    >
      <div
        class="c1"
        type="about-wrapper"
      >
        <img
          alt="About image"
          class="c2"
          loading="lazy"
          src="/images/casual-image.webp"
        />
      </div>
      <div
        class="c3"
        type="text-container-about"
      >
        <span
          class="c4"
          from="#1E90FF"
          to="#00008B"
          via="#4169E1"
        >
          <h2
            class="c5"
          >
            Sobre mim
          </h2>
        </span>
        <p
          class="c6"
          color="#FFFFFF"
          type="about"
        >
          Sou desenvolvedor full stack com uma paixão por criar aplicações web interativas e responsivas. Tenho experiência com JavaScript, React, Redux, Node.js, Express, PostgreSQL, Sequelize, HTML, CSS, Next.js, Laravel e Git. Sou um aprendiz rápido e estou sempre buscando expandir meu conhecimento e habilidades. Destaco-me por minha comunicação eficaz, trabalho em equipe, resolução de problemas, adaptabilidade, gestão de tempo, pensamento crítico, inovação e criatividade, liderança, empatia e inteligência emocional, além de uma forte ética profissional.
        </p>
        <div
          class="c7"
          type="tab-container"
        >
          <button
            class="c8 dark"
            color="#1E90FF"
            from="#1E90FF"
            primarycolor="#FFFFFF"
            to="#00008B"
            via="#4169E1"
          >
            Hard Skills
          </button>
          <button
            class="c9 dark"
            color="#1E90FF"
            from="#1E90FF"
            primarycolor="#FFFFFF"
            to="#00008B"
            via="#4169E1"
          >
            Soft Skills
          </button>
          <button
            class="c9 dark"
            color="#1E90FF"
            from="#1E90FF"
            primarycolor="#FFFFFF"
            to="#00008B"
            via="#4169E1"
          >
            Educação
          </button>
        </div>
        <div
          class="c10"
          color="#FFFFFF"
        >
          <ul
            class="c11 dark"
          >
            <li>
              Node.js
            </li>
            <li>
              Express
            </li>
            <li>
              MySQL
            </li>
            <li>
              Sequelize
            </li>
            <li>
              JavaScript
            </li>
            <li>
              Next.js
            </li>
            <li>
              Laravel
            </li>
          </ul>
        </div>
      </div>
    </section>
  </div>
</div>
`;
